cmd/spire-server/cli/run/run.go

```diff
@@ -525,6 +549,67 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 	return sc, nil
 }
 
+func parseBundleEndpointProfile(config federatesWithConfig) (trustDomainConfig *bundleClient.TrustDomainConfig, err error) {
+	var data bytes.Buffer
+	if err := printer.DefaultConfig.Fprint(&data, config.BundleEndpointProfile); err != nil {
+		return nil, err
+	}
+	configString := data.String()
+	profileConfig := new(bundleEndpointProfileConfig)
+	if err := hcl.Decode(profileConfig, configString); err != nil {
+		return nil, fmt.Errorf("failed to decode configuration: %w", err)
+	}
+
+	var endpointProfile bundleClient.EndpointProfileInfo
+	switch {
+	case profileConfig.HTTPSWeb != nil:
+		endpointProfile = bundleClient.HTTPSWebProfile{}
+	case profileConfig.HTTPSSPIFFE != nil:
+		spiffeID, err := spiffeid.FromString(profileConfig.HTTPSSPIFFE.EndpointSPIFFEID)
+		if err != nil {
+			return nil, fmt.Errorf("could not get endpoint SPIFFE ID: %w", err)
+		}
+		endpointProfile = bundleClient.HTTPSSPIFFEProfile{EndpointSPIFFEID: spiffeID}
+	default:
+		return nil, errors.New("no bundle endpoint profile defined")
+	}
+
+	return &bundleClient.TrustDomainConfig{
+		EndpointURL:     config.BundleEndpointURL,
+		EndpointProfile: endpointProfile,
+	}, nil
+}
+
+func parseDeprecatedBundleEndpoint(config *deprecatedFederatesWithBundleEndpointConfig) (trustDomainConfig *bundleClient.TrustDomainConfig, err error) {
+	port := defaultBundleEndpointPort
+	if config.Port != 0 {
+		port = config.Port
+	}
+	if config.UseWebPKI && config.SpiffeID != "" {
+		return nil, errors.New("usage of `bundle_endpoint.spiffe_id` is not allowed when authenticating with Web PKI")
+	}
+
+	var endpointProfile bundleClient.EndpointProfileInfo
+	if config.UseWebPKI {
+		endpointProfile = bundleClient.HTTPSWebProfile{}
+	} else {
+		var spiffeID spiffeid.ID
+		if config.SpiffeID != "" {
+			spiffeID, err = spiffeid.FromString(config.SpiffeID)
+			if err != nil {
+				return nil, err
```

Comment:
```suggestion
				return nil, fmt.Errorf("Could not parse endpoint SPIFFE ID %q: %w", config.SpiffeID, err)
```

Similar comment as above, TD name here might help diagnose this error.

Reply:
The caller to this function already includes the trust domain as part of the error description. This error is embedded in the error returned by the caller.